### PR TITLE
Feature/16 audit trail service setup

### PR DESCRIPTION
**Pull request by FahimDev**
- Creates NestJs project as Audit Trail Service
- Adds ENV Example file 
- Configures Docker 
- Dockerize Backend
- Dockerize DB
- Dockerize PgAdmin

### CODE REVIEW

File: backend/audit-trail-service/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - 5432:5432
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
+
+  pgadmin:
+    container_name: pgadmin4
+    image: dpage/pgadmin4:8.0
+    environment:
+      PGADMIN_DEFAULT_EMAIL: admin@admin.com
+      PGADMIN_DEFAULT_PASSWORD: root
```
Comment (dipanjal) [inline backend/audit-trail-service/docker-compose.yml:42]: avoid putting credentials directly. It's good to use `env_files`

File: backend/audit-trail-service/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ports:
+      - '5050:80'
+    depends_on:
+      - pgsqldb
+
+volumes:
+  postgres-data:
```
Comment (dipanjal) [inline backend/audit-trail-service/docker-compose.yml:49]: add a newline at EOF

**Review by dipanjal — CHANGES_REQUESTED** [review]
Please resolve the requested changes

File: backend/audit-trail-service/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all"
+}
```
[the remark this answers is not in the document]
Comment (FahimDev) [reply backend/audit-trail-service/.prettierrc:4]: I have changed the other files. But this one is generated by the framework. So, I skipped this file.

File: backend/audit-trail-service/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    environment:
+      APP_ENV: ${APP_ENV}
+      APP_PORT: ${APP_PORT}
+      DB_HOST: postgres-db
+      DB_PORT: ${DB_PORT}
+      DB_NAME: ${DB_NAME}
+      DB_USER: ${DB_USER}
+      DB_PASS: ${DB_PASS}
+    env_file:  # Add this line to specify the environment file
+      - .env  # Specify the path to your .env file
```
[the remark this answers is not in the document]
Comment (FahimDev) [reply backend/audit-trail-service/docker-compose.yml:23]: I was following the references of .gitignore file also some other article where they have suggested to use the type name after the .env file. So I renamed by file as .env.local

File: backend/audit-trail-service/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    volumes:
+      - ./:/usr/src/app
+    env_file:  # Add this line to specify the environment file
+      - .env.local # Specify the path to your .env file
+
+  pgsqldb:
+    container_name: pgsqldb
+    image: postgres:16
+    env_file:
+      - .env.local
```
Comment (dipanjal) [inline backend/audit-trail-service/docker-compose.yml:20]: It's standard to use separate and specific env file for database. Ex: database.env
Try not to mix backend env with database env

File: backend/audit-trail-service/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    ports:
+      - 5432:5432
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
+
+  pgadmin:
+    container_name: pgadmin4
+    image: dpage/pgadmin4:8.0
+    env_file:
+      - .env.local
```
Comment (dipanjal) [inline backend/audit-trail-service/docker-compose.yml:30]: Same as previous

**Review by dipanjal — APPROVED** [review]
Left a minor comment, though it's not super urgent but one of the best practices

File: backend/audit-trail-service/docker-compose.yml (hunk 6 above)
Comment (FahimDev) [reply backend/audit-trail-service/docker-compose.yml:30]: Understood!

File: backend/audit-trail-service/docker-compose.yml (hunk 5 above)
Comment (FahimDev) [reply backend/audit-trail-service/docker-compose.yml:20]: Understood!

**Pull request merged**